src/include/processor/operator/update/delete_manager.h

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+#include "processor/execution_context.h"
+#include "processor/result/result_set.h"
+#include "storage/store/node_table.h"
+
+namespace kuzu {
+namespace processor {
+
+class NodeDeleteManager {
```

Comment:
I wouldn't call this Manager, maybe executor is a more appropriate suffix. Another term in my mind is `NodeDeleter`. Not sure if this is a valid word. You might need to take a further look.